### ISSUE TITLE
Handling arbitrary graphemes

### ISSUE THREAD

**Issue by Anders429** [Anders429/word_filter#20]
Suppose we want to make a `WordFilter` that filters the word "bar". It's pretty straight-forward:

```rust
use word_filter::WordFilterBuilder;

let filter = WordFilterBuilder::new().words(&["bar"]).build();
```

With the above definition, we can easily check for "bar":

```rust
assert!(filter.check("bar"));
```

However, suppose a user wants to get around this check. They add a diacritical mark and get around the filter:

```rust
assert!(filter.check("bãr"))  // Oh no!
```

Now, to combat this, we try to add an alias:
```rust
use word_filter::WordFilterBuilder;

let filter = WordFilterBuilder::new().words(&["bar"]).aliases(&[("a", "ã")]).build();
```

but nothing stops the user from adding more diacritical marks: 
```rust
assert!(filter.check("bà̵̡͚̎̍͘r"))  // Oh no!!
```

No amount of aliases will ever stop users from circumventing filters arbitrarily in this way. 

Note also that there is a difference between "ã" and "ã". The first is one character, while the second is actually two. 

Here lies the problem. How can we possibly match every single possible grapheme with the same base character? We can't. Thus, having graphemes as edges is actually *not* desirable. 

I'm not sure the best way to handle this. We could go back to character edges, but that prevents graphemes from being used in words and exceptions properly, as they won't be repeated properly in matching. We could try to detect when a character is a combining diacritical mark (as opposed to a base character) and create an exception for callbacks and targets when matching, but I'm not sure how practical that is. 

Turns out, unicode is complicated lol.

**Comment by Anders429** [comment]
Ideally, the best solution would be one that accepts graphemes in build inputs (words or exceptions), but also allows for ignoring them in runtime inputs (text to be filtered). 

Something like this could work: 
- Switch to using `char`s as node edges.
- Creating subgraphs that contain the grapheme cluster's chars as nodes and set the return node to return to the rest of the word/exception. Note that these nodes will need to be owned, likely by the node that calls into them.
- For matching added diacritical marks, it will be up to the user to add those as valid separators. Returning to character-based matching will allow for this to be done easily.

This will allow repeated grapheme clusters to be matched correctly, while also resolving the arbitrary grapheme cluster issue.

**Comment by Anders429** [comment]
This will be fixed with #22. Arbitrary graphemes can be handled by letting undesired non-spacing marks be valid separators.
